lib/fog/libvirt/models/compute/volume.rb

```diff
@@ -71,6 +72,7 @@ def clone_volume(new_name)
           new_volume.path = service.clone_volume(pool_name, new_volume.to_xml, self.name).path
 
           new_volume.reload
+          new_volume
```

Comment:
I'm don't think this is correct, during modifying the tests i discovered that `new_volume.reload` wasn't returning a volume here and I suspect I'm missing understanding something around the mocks behaviour as to why reload returns nil here when I adjust what volumes are returned.

Same for the above, I just modified both to be consistent.

Reply:
I believe reload will basically do `volumes.get(new_volume.identity)` in this context and should return the result of that call. You can see that here: https://github.com/fog/fog-core/blob/7865ef77ea990fd0d085e49c28e15957b7ce0d2b/lib/fog/core/model.rb#L44

Does that help clarify what the expectations would be? I'm not familiar with the current state of the mocks and things, but hope this might help point you in the right direction. I'm certainly happy to answer other questions as you have them also.

Reply:
Thanks, I think it helped give me the hint that reload was calling list_volumes with a filter and that allowed me to work out that reload will trigger a look up of the volume by `key` using the volume `:id` attribute. So in the mocks below, to support filtering it is necessary to ensure `:id` and `:key` have the same value.

Reply:
Great, glad that helped nudge you in the right direction.